include/tvm/runtime/relax_vm/memory_manager.h

```diff
@@ -71,6 +72,14 @@ class Allocator {
    *  \return A sized allocation in the form of a buffer.
    */
   virtual Buffer Alloc(size_t nbytes, size_t alignment, DLDataType type_hint) = 0;
+  /*! \brief Allocate a buffer given a size, alignment and type.
+   * \param ndim The number of dimension of allocated tensor.
+   * \param shape The shape of allocated tensor.
+   *  \param type_hint A type hint to the allocator.
+   * \param mem_scope The memory scope of allocated tensor.
+   *  \return A sized allocation in the form of a buffer.
+   */
+  virtual Buffer Alloc(int ndim, const int64_t* shape, DLDataType type_hint, String mem_scope);
```

Comment:
Let us change int ndim, const int64_t* shape to ShapeTuple

Reply:
Thanks, I've updated the API with `ShapeTuple`.

Reply:
Actually I just realized that we could probably unify the 2 Alloc overloads into one and call the appropriate AllocDataSpace based on the size of the ShapeTuple. Was that intended change mentioned here?

Reply:
let us keep things in two for now as there might be some future thoughts on byte reuse, as the original alloca is dtype agnostic